javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -563,26 +564,53 @@ private[net] trait InetAddressBase {
   }
 }
 
-object InetAddress extends InetAddressBase
+object InetAddress extends InetAddressBase {
+  // cached host values are discarded after this amount of time (seconds)
+  private val HostTimeout: Int =
+    sys.props
+      .get("networkaddress.cache.ttl")
+      .map(_.toInt)
+      .getOrElse(30)
+
+  // failed lookups are retried after this amount of time (seconds)
+  private val NegativeHostTimeout: Int =
+    sys.props
+      .get("networkaddress.cache.negative.ttl")
+      .map(_.toInt)
+      .getOrElse(10)
+}
 
 class InetAddress private[net] (
     ipAddress: Array[Byte],
     private var host: String
 ) extends Serializable {
   import InetAddress._
 
+  private var hostLastAccessed: time_t = 0
+  private var lastLookupFailed = false
+
   private[net] def this(ipAddress: Array[Byte]) = this(ipAddress, null)
 
   def getHostAddress(): String = createIPStringFromByteArray(ipAddress)
 
+  private def hostTimeoutExpired(timeNow: time_t): Boolean = {
+    val timeout = if (lastLookupFailed) NegativeHostTimeout else HostTimeout
+    difftime(timeNow, hostLastAccessed) > timeout
+  }
+
   def getHostName(): String = {
-    if (host == null) {
+    val timeNow = time(null)
+    if (host == null || hostTimeoutExpired(timeNow)) {
+      hostLastAccessed = timeNow
       val ipString = createIPStringFromByteArray(ipAddress)
-      host = SocketHelpers
-        .ipToHost(ipString, isValidIPv6Address(ipString))
-        .getOrElse {
-          ipString
-        }
+      SocketHelpers.ipToHost(ipString, isValidIPv6Address(ipString)) match {
+        case None =>
+          lastLookupFailed = true
+          host = ipString
```

Comment:
In case if lookup would fail we should not set ipString here, if the hostname was passed as an constructor argument and is was null it would override it. In some cases lookup of `localhost` was failing (it was on my machine), so it would permanently remove the original host info. 
We should remove this line and make some additional changes in the logic: 
 - We should rename a `host` to `originalHost` and make it immutable, it should be always returned if present as pointed by java docs `If this InetAddress was created with a host name, this host name will be remembered and returned; otherwise, a reverse name lookup will be performed and the result will be returned based on the system configured name lookup service.`
 - We should create a new variable that would keep the hostname after DNS lookup and use it for timed caching. 
```suggestion
```

This should resolve the problems seen in the CI (tested locally with minimal change of removing the offending line)

Reply:
Sorry for not paying enough attention to the javadocs :sweat_smile: It should be fixed now. Thanks for the guidance!